src/core/asyncUtils.ts

```diff
@@ -1,28 +1,15 @@
-import { act } from 'react-test-renderer'
+import { Act, WaitOptions, AsyncUtils } from '../types'
 
-export interface WaitOptions {
-  interval?: number
-  timeout?: number
-  suppressErrors?: boolean
-}
-
-class TimeoutError extends Error {
-  constructor(util: Function, timeout: number) {
-    super(`Timed out in ${util.name} after ${timeout}ms.`)
-  }
-}
-
-function resolveAfter(ms: number) {
-  return new Promise((resolve) => {
-    setTimeout(resolve, ms)
-  })
-}
+import { resolveAfter } from '../helpers/promises'
+import { TimeoutError } from '../helpers/error'
 
-function asyncUtils(addResolver: (callback: () => void) => void) {
+function asyncUtils(act: Act, addResolver: (callback: () => void) => void): AsyncUtils {
   let nextUpdatePromise: Promise<void> | null = null
 
   const waitForNextUpdate = async ({ timeout }: Pick<WaitOptions, 'timeout'> = {}) => {
-    if (!nextUpdatePromise) {
+    if (nextUpdatePromise) {
+      await nextUpdatePromise
+    } else {
```

Comment:
small refactor here.  in the old code, we were double `await`ing the same promise.  Minor issue with almost zero impact in practice, but made debugging the async test issues I saw when trying to implement a preact renderer more difficult than it should have been